src/lib/auth.ts

```diff
@@ -4,13 +4,23 @@ export type AccessContext = {
   db: typeof db;
   user: { id: string };
   workspaceId: string | null;
+  headers: Headers;
 };
 
 export const checkWorkspaceAccess = async (
   ctx: AccessContext,
   workspaceIdOfTheResource: string,
 ) => {
-  if (ctx.workspaceId && workspaceIdOfTheResource !== ctx.workspaceId) {
+  const referer = ctx.headers.get("referer");
+  const refererUrl = referer ? new URL(referer) : null;
+  const zapierInReferer = refererUrl?.searchParams
+    .get("redirect_uri")
+    ?.includes("zapier");
```

Comment:
'referer' -> 'referrer'

Reply:
It's actually 'referer' for http headers. see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer